extensions/git/src/repository.ts

```diff
@@ -844,21 +844,70 @@ export class Repository implements Disposable {
 				}
 			});
 
-			const promises: Promise<void>[] = [];
+			const maxCommandLineLength: number = 30000;
```

Comment:
The limit on windows seems to be around 31k characters, so for now I chose 30k as a nice round number.  The total length of the command line arguments will be something over the length of the joined files passed to the repository command based on other arguments added, such as `checkout -q -- <files>` in the case of the git repository.

Unix systems can potentially support much larger command line lengths, so this chunking size may be unnecessarily small for such systems, so it may be better to check the actual allowable chunk size for the system being ran on, than to use a hardcoded value such as this.  `xargs --show-limits` can report the upper limit on command length on a unix, or most unix-like systems.

Reply:
As a note, in the issue you mentioned that

> As a first attempt, the code could take 4096 as an upper limit, which is the minimum allowable max length of a unix system, as reported by xargs, and is the smallest common denominator?

So if someone has that configured we could still get another Issue for vscode about the 30k not being small enough. This limitation is quite frustrating. I would love to know why they have these constraints in place.